src/chains/filecoin/filecoin/src/api.ts

```diff
@@ -1,79 +1,166 @@
+  }
+
+  // Reference implementation entry point: https://git.io/JtO3a
+  "Filecoin.ChainNotify"(rpcId?: string): PromiEvent<Subscription> {
+    const subscriptionId = this.#getId();
+    let promiEvent: PromiEvent<Subscription>;
+
+    const currentHead = new HeadChange({
+      type: HeadChangeType.HCCurrent,
+      val: this.#blockchain.latestTipset()
+    });
+
+    const unsubscribeFromEmittery = this.#blockchain.on(
+      "tipset",
+      (tipset: Tipset) => {
+        // Ganache currently doesn't support Filecoin reorgs,
+        // so we'll always only have one tipset per head change
+        // See reference implementations here: https://git.io/JtOOk;
+        // other lines of interest are line 207 which shows only the chainstore only
+        // references the "hcnf" (head change notification function) in the
+        // reorgWorker function (lines 485-560)
+
+        // Ganache currently doesn't support Filecoin reverts,
+        // so we'll always use HCApply for now
+
+        const newHead = new HeadChange({
+          type: HeadChangeType.HCApply,
+          val: tipset
+        });
+
+        if (promiEvent) {
+          promiEvent.emit("message", {
+            type: SubscriptionMethod.ChannelUpdated,
+            data: [subscriptionId.toString(), [newHead.serialize()]]
+          });
+        }
+      }
+    );
+
+    const unsubscribe = (): void => {
+      unsubscribeFromEmittery();
+      // Per https://git.io/JtOc1 and https://git.io/JtO3H
+      // implementations, we're should cancel the subscription
+      // since the protocol technically supports multiple channels
+      // per subscription, but implementation seems to show that there's
+      // only one channel per subscription
+      if (rpcId) {
+        promiEvent.emit("message", {
+          type: SubscriptionMethod.SubscriptionCanceled,
+          data: [rpcId]
+        });
+      }
+    };
+
+    promiEvent = PromiEvent.resolve({
+      unsubscribe,
+      id: subscriptionId
+    });
+
+    // There currently isn't an unsubscribe method,
+    // but it would go here
+    this.#subscriptions.set(subscriptionId.toString()!, unsubscribe);
+
+    promiEvent.emit("message", {
+      type: SubscriptionMethod.ChannelUpdated,
+      data: [subscriptionId.toString(), [currentHead.serialize()]]
+    });
+
+    return promiEvent;
+  }
+
+  [SubscriptionMethod.ChannelClosed](
+    subscriptionId: SubscriptionId
+  ): Promise<boolean> {
+    const subscriptions = this.#subscriptions;
+    const unsubscribe = this.#subscriptions.get(subscriptionId);
+
+    if (unsubscribe) {
+      subscriptions.delete(subscriptionId);
+      unsubscribe();
+      return Promise.resolve(true);
+    } else {
+      return Promise.resolve(false);
+    }
   }
 
-  async "Filecoin.StateListMiners"(): Promise<Array<SerializedMiner>> {
-    return [this[_blockchain].miner.serialize()];
+  async "Filecoin.StateListMiners"(): Promise<Array<string>> {
+    return [this.#blockchain.miner];
```

Comment:
No because `miner` was changed to type `string`. If you saw, I removed the `Miner` type you had before because it's not a real Filecoin API type. ID protocol address support was added in a distant branch last night to change the behavior of this: https://github.com/trufflesuite/ganache-core/commit/b67757aab542868fa8e5f00dac21b8a1bda21e63

Reply:
Sounds good!